src/tree/updater_quantile_hist.h

```diff
@@ -409,6 +408,10 @@ class QuantileHistMaker: public TreeUpdater {
     common::ColumnSampler column_sampler_;
     // the internal row sets
     RowSetCollection row_set_collection_;
+    // tree rows that were not used for current training
+    std::vector<size_t> unused_rows_;
+    // feature vectors for subsampled prediction
+    std::vector<RegTree::FVec> feat_vecs_;
```

Comment:
Do you get any performance benefit from keeping this as a member variable instead of creating locally? Avoid extra member variables if possible.

Reply:
Yes, this field allows us not to allocate `nthread*nfeatures` units of memory on each `UpdatePredicionCache()` call. Here're the results for one of our datasets:

Santander, subsample == 0.5 | UpdatePredictionCache, s | Overall time, s
-- | -- | --
with `feat_vecs_` as a member variable | 42.7465 | 142.7305
with local allocations | 49.3696 | 147.4053